plugins/analytics/public/dockerhub.py

```diff
@@ -0,0 +1,201 @@
+import json
+import logging
+import re
+
+import requests
+from core import taskmanager
+from core.config.config import yeti_config
+from core.schemas import observable, task
+from core.schemas.observable import Observable, ObservableType
+
+
+class DockerHubApi(object):
+    """Base class for querying the DockerHub API."""
+
+    @staticmethod
+    def _make_request(endpoint) -> dict:
+        response = requests.get(endpoint, allow_redirects=True)
+        if response:
+            return response.json()
+        return {}
+
+    @staticmethod
+    def _iter_endpoint_pages(endpoint) -> dict:
+        data = DockerHubApi._make_request(endpoint)
+        while data:
+            for result in data.get("results", []):
+                yield result
+            next = data.get("next")
+            if next:
+                data = DockerHubApi._make_request(next)
+            else:
+                data = {}
+
+    @staticmethod
+    def inspect_image(image, tag=None) -> dict:
+        if tag:
+            endpoint = (
+                f"https://hub.docker.com/v2/repositories/{image}/tags/{tag}/images"
+            )
+        else:
+            endpoint = f"https://hub.docker.com/v2/repositories/{image}"
+        return DockerHubApi._make_request(endpoint)
+
+    @staticmethod
+    def inspect_user(user) -> dict:
+        endpoint = f"https://hub.docker.com/v2/orgs/{user}"
+        # if orgs does not exist, redirects to
+        # https://hub.docker.com/v2/users/
+        return DockerHubApi._make_request(endpoint)
+
+    @staticmethod
+    def user_images(user, page_size=50) -> iter:
+        endpoint = f"https://hub.docker.com/v2/repositories/{user}?page_size={page_size}&ordering=last_updated"
+        yield from DockerHubApi._iter_endpoint_pages(endpoint)
+
+    # https://hub.docker.com/v2/repositories/yetiplatform/yeti/tags/2.0.1
```

Comment:
What's this comment supposed to address?

Reply:
Forgot to remove it... Fixed in [86f12fb](https://github.com/yeti-platform/yeti/pull/904/commits/86f12fb93141cb7a959c20f6a1201c5c1fdc490f)